### ISSUE TITLE
price hit variation - closing/open price alarm

### ISSUE THREAD

**Issue by algazarra** [drpout/boilr#161]
sometimes a trader only cares for open/closing price as it will define trend. it would be interesting to have a spin off of the price hit alarm.

settings would be something like:
if price close >/</= X RING

**Comment by dllud** [comment]
Cryptocurrency markets do not open and close, they are live 24/7. Therefore either:
a) we hardcode an open/close date-time into Boilr;
b) we let the user set his own definition of open/close date-time.
How should we do it?

The way you define the open/close date-time depends on the interval you are tracking:
- Monthly: you need day of the month and time (e.g. day 1 at 00:00).
- Weekly: day of the week and time (e.g. Monday at 00:00).
- Daily: time (e.g. 00:00)

For sub-daily intervals (e.g. 30 min) it's logic to follow something like `00:00, 00:30, 01:00, ...` and not `00:07, 00:37, 01:07`. Thus there would be no need for a user setting on those.

**Comment by algazarra** [comment]
sorry, I meant close of the candlestick! but you still raise a valid question, what timeframe should be considered? users would have to define it

**Comment by dllud** [comment]
LOL, close of the candlestick? That is a date-time hardcoded into each charting library. We could do the same and thus go for option a) above. Though I think it's better to go for b), I bet not everyone likes their daily open/close at 00:00.

**Comment by algazarra** [comment]
b) seems better, coinigy.com has a cool feature where you specify the timeframe

![2015-04-07-165453_1680x999_scrot](https://cloud.githubusercontent.com/assets/7598320/7027343/dddbd660-dd46-11e4-8d89-d2065dec3d3a.png)

you input a number followed by D for day, M for month, etc.

I think I suggested this before. Anyway, I don't like the method we currently use (for example on price change alarms), where you have to calculate in minutes. it would be great if we could work the same way as coinigy. So this is a double feature request! wan't me to open another issue?

**Comment by dllud** [comment]
b) has nothing to do with timeframe, b) is about letting the user set his own definition of open/close date-time for the selected timeframe.

Being able to select the timeframe was never on stake.

Regarding timeframe input, please refer to #60 where **you** found a limitation on that Coinigy approach (being unable set 1.5 min) and then we agreed on a better solution using a HMS Picker.

**Comment by algazarra** [comment]
>b) is about letting the user set his own definition of open/close date-time for the selected timeframe
I'm not sure I understand. Candlestick's open and close are defined by timeframe and timezone alone right? call me on csip, lets talk it through.

>Regarding timeframe input, please refer to #60 where you found a limitation on that Coinigy approach (being unable set 1.5 min) and then we agreed on a better solution using a HMS Picker.
It's true, it has that limitation, but the easy workaround is simply using the next smaller scale. in this case you would input 90s.

As for the picker on #60, it presents so many problems and it doesn't insure much user experience improvement, that I think it would be like killing an ant with a bazooka

**Comment by dllud** [comment]
Agreed with @algazarra to go for option b). Therefore this alarm will have two time settings:
- timeframe (in minutes), which is roughly the alarm update interval.
- offset (in minutes), the offset of the timeframe in relation to its base starting point (probably 00:00 UTC on the 1st of January of current year).

Minutes input interface will be replaced by a HMS picker as described on #60.

When timeframe equals:
- 1 day (1440 min), offset will be replaced by a more friendly input with a time zones list.
- 1 week, offset replaced by a days of the week list, i.e. you'll choose on which day the week starts, e.g. Sunday or Monday.
